doc/api/index.rst

```diff
@@ -1,67 +1,45 @@
-API Overview
-============
+API
+===
 
 .. toctree::
    :hidden:
 
    api_changes
 
-.. contents:: :local:
+For recent changes, see :doc:`api_changes`.
 
-See also the :doc:`api_changes`.
+When using the library you will typically create
+:doc:`Figure <figure_api>` and :doc:`Axes <axes_api>` objects and 
+call their methods to add content and modify the apprearance.
 
-Usage patterns
---------------
-
-Below we describe several common approaches to plotting with Matplotlib.
-
-The pyplot API
-^^^^^^^^^^^^^^
-
-`matplotlib.pyplot` is a collection of command style functions that make
-Matplotlib work like MATLAB. Each pyplot function makes some change to a
-figure: e.g., creates a figure, creates a plotting area in a figure, plots
-some lines in a plotting area, decorates the plot with labels, etc.
+  - :doc:`figure_api`: axes creation, figure-level content
+  - :doc:`axes_api`: most plotting methods, Axes labels, access to 
+    axis styling, etc.  
 
-`.pyplot` is mainly intended for interactive plots and simple cases of
-programmatic plot generation.
+Example: We create a Figure ``fig`` and Axes ``ax``. Then we call
+methods on them to plot data, add axis labels and a figure title.
+ 
+.. plot::
+   :include-source:
+   :align: center
 
-Further reading:
+   import matplotlib.pyplot as plt
+   import numpy as np
 
-- The `matplotlib.pyplot` function reference
-- :doc:`/tutorials/introductory/pyplot`
-- :ref:`Pyplot examples <pyplots_examples>`
+   x = np.arange(0, 4, 0.1)
+   y = np.sin(x*np.pi)
 
-.. _api-index:
+   fig, ax = plt.subplots(figsize=(3,2), constrained_layout=True)
+   ax.plot(x, y)
+   ax.set_xlabel('t [s]')
+   ax.set_ylabel('S [V]')
+   fig.suptitle('Sine wave')
```

Comment:
```suggestion
   ax.set_title('Sine wave')
   fig.patch.set_color('lightsteelblue')
```
One would not use `fig.suptitle` with a single Axes. This looks a bit odd because the title is centered on the figure not the Axes. There are actually not too many reasonable uses for figure methods in this simple example. Coloring the background may be one of the better ones.

Reply:
Are we trying to soft deprecate `fig.set_facecolor`?  I understand not wanting duplicate ways of doing the same thing, but asking folks to modify a property like this is a discoverability issue.